### PR TITLE
MGMT-15243: Skip any zero size manifests when applying

### PR DESCRIPTION
**Pull request by paul-maidment**
When applying manifests when installation is started, we should skip any
manifests that are of zero size to prevent them from causing bootstrap
problems later.

We can't validate zero size at the point of file upload as the UI is
presently using an approach where it uploads an empty manifest file to
track that the user wishes to add user defined manifests.

Discussions are underway to change this but for now we need to take this
alternative approach.

This PR implements this.

## List all the issues related to this PR

- [ ] New Feature 
- [x] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [x] Automation (CI, tools, etc)
- [x] Cloud
- [x] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/ignition/ignition.go [hunk 1]
```diff
@@ -1565,6 +1565,10 @@ func (g *installerGenerator) runCreateCommand(ctx context.Context, installerPath
 	return nil
 }
 
+type Item struct {
```
Comment (carbonin) [inline internal/ignition/ignition.go:1568]: What is this?

File: internal/ignition/ignition.go [hunk 2]
```diff
@@ -1574,6 +1578,12 @@ func (g *installerGenerator) downloadManifest(ctx context.Context, manifest stri
 	if err != nil {
 		return err
 	}
+
+	if len(content) == 0 {
+		// Ignore any empty files.
+		return nil
```
Comment (carbonin) [inline internal/ignition/ignition.go:1584]: I'd prefer for this not to ever actually be present.
Why can't we validate this on the way in and tell the user they're doing something wrong?

**Review by carbonin — CHANGES_REQUESTED** [review]

File: internal/ignition/ignition.go (hunk 2 above)
Comment (paul-maidment) [reply internal/ignition/ignition.go:1584]: Sorry, you mean on upload?

File: internal/ignition/ignition_test.go [hunk 3]
```diff
@@ -921,15 +921,29 @@ var _ = Describe("downloadManifest", func() {
...
+		manifestName := fmt.Sprintf("%s/manifests/openshift/masters-chrony-configuration.yaml", cluster.ID)
+		mockS3Client.EXPECT().Download(ctx, manifestName).Return(io.NopCloser(strings.NewReader("")), int64(0), nil)
+		Expect(os.Mkdir(filepath.Join(workDir, "/openshift"), 0755)).To(Succeed())
+		Expect(os.Mkdir(filepath.Join(workDir, "/manifests"), 0755)).To(Succeed())
+
+		Expect(generator.downloadManifest(ctx, manifestName)).To(Succeed())
+
+		_, err := os.Stat(filepath.Join(workDir, "/openshift/masters-chrony-configuration.yaml"))
+		Expect(err).To(HaveOccurred())
+		Expect(err.Error()).To(ContainSubstring("no such file or directory"))
```
Comment (carbonin) [inline internal/ignition/ignition_test.go:946]: I think the better way to do this is `errors.Is(err, fs.ErrNotExist)`.

String matching is generally not a good choice.

**Review by carbonin — APPROVED** [review]

**Pull request merged**